bookkeeper-server/src/main/java/org/apache/bookkeeper/client/EnsemblePlacementPolicy.java

```diff
@@ -107,7 +137,24 @@ public EnsemblePlacementPolicy initialize(ClientConfiguration conf,
      * @throws BKNotEnoughBookiesException
      * @return the org.apache.bookkeeper.net.BookieSocketAddress
      */
-    public BookieSocketAddress replaceBookie(int ensembleSize, int writeQuorumSize, int ackQuorumSize, java.util.Map<String, byte[]> customMetadata, Collection<BookieSocketAddress> currentEnsemble, BookieSocketAddress bookieToReplace, Set<BookieSocketAddress> excludeBookies) throws BKNotEnoughBookiesException;
+    public default BookieSocketAddress replaceBookie(int ensembleSize, int writeQuorumSize, int ackQuorumSize,
+        java.util.Map<String, byte[]> customMetadata, Collection<BookieSocketAddress> currentEnsemble,
+        BookieSocketAddress bookieToReplace, Set<BookieSocketAddress> excludeBookies) throws BKNotEnoughBookiesException {
+        return replaceBookie(bookieToReplace, excludeBookies);
+    }
+
+    /**
+     * Legacy method introduce for compatibility with 4.4 clients
+     * @param bookieToReplace bookie to replace
+     * @param excludeBookies bookies that should not be considered as candidate.
+     * @return the org.apache.bookkeeper.net.BookieSocketAddress
+     * @throws org.apache.bookkeeper.client.BKException.BKNotEnoughBookiesException
+     * @see #replaceBookie(int, int, int, java.util.Map, java.util.Collection, org.apache.bookkeeper.net.BookieSocketAddress, java.util.Set)
+     */
+    public default BookieSocketAddress replaceBookie(BookieSocketAddress bookieToReplace,
```

Comment:
same comment as above.

- the default behavior for the new method throws UnsupportedOperationException
- the legacy method should call the new method by default.